src/document/url.ml

```diff
@@ -87,13 +87,22 @@ module Path = struct
   let mk ?parent kind name = { kind ; parent ; name }
 
   let rec from_identifier : source -> t = function
-    | `Root (abstr, unit_name) ->
-      let parent = mk "package" abstr.Root.package in
+    | `Root (parent, unit_name) ->
+      let parent = from_identifier (parent :> source) in
       let kind = "module" in
       let page = ModuleName.to_string unit_name in
       mk ~parent kind page
-    | `Page (abstr, page_name) ->
-      let parent = mk "package" abstr.Root.package in
+    | `RootPage page_name ->
+      let kind = "cpage" in
+      let page = PageName.to_string page_name in
+      mk kind page
+    | `Page (parent, page_name) ->
+      let parent = from_identifier (parent :> source) in
+      let kind = "cpage" in
+      let page = PageName.to_string page_name in
+      mk ~parent kind page
+    | `LeafPage (parent, page_name) ->
+      let parent = from_identifier (parent :> source) in
```

Comment:
What's this "cpage" kind ? You can use something more longformed (the content is pretty arbitrary).

Reply:
'container page' - it was originally that, but it appeared everywhere in the latex docs so I cut it down to this. 'container pages' are those that can contain others (Page and RootPage), and are output in the html as <pagename>/index.html, in contrast to the 'page' types that are output as <pagename>.html

Reply:
I see, and we don't know if they have children or not just by looking at their urls.

I would suggest just using `container-page`. We already have `module-type`, and it's fine.